script/script.go

```diff
@@ -494,12 +558,23 @@ func setRequestHeader(f filters.FilterContext) func(*lua.LState) int {
 	}
 }
 
+func iterateRequestHeader(f filters.FilterContext) func(*lua.LState) int {
+	// https://www.lua.org/pil/7.2.html
+	return func(s *lua.LState) int {
+		s.Push(s.NewFunction(nextHeader(f.Request().Header)))
+		return 1
+	}
+}
+
 func getResponseHeader(f filters.FilterContext) func(*lua.LState) int {
 	return func(s *lua.LState) int {
 		hdr := s.ToString(-1)
 		res := f.Response().Header.Get(hdr)
-		s.Push(lua.LString(res))
-		return 1
+		if res != "" {
```

Comment:
Here is also a breaking change. Before it would return empty string if header is missing and now it returns `nil` which is usual in lua return value from table when key is missing.

Reply:
Let's add a TODO as commented in https://github.com/zalando/skipper/pull/1485#discussion_r453818921